modules/cli/src/main/scala/scala/cli/commands/InstallHome.scala

```diff
@@ -4,64 +4,96 @@ import caseapp._
+    val binDirPath =
+      options.binDirPath.getOrElse(scala.build.Directories.default().binRepoDir / "scala-cli")
+    val newScalaCliBinPath = os.Path(options.scalaCliBinaryPath, os.pwd)
+
+    val newVersion: String =
+      os.proc(newScalaCliBinPath, "about", "--version").call(cwd = os.pwd).out.text.trim
 
-    if (os.exists(scalaCliBinPath)) {
-      if (options.force) os.remove.all(scalaCliBinPath)
-      else if (coursier.paths.Util.useAnsiOutput()) {
-        println(
-          "scala-cli already exists. Do you want to override it [Y/n]"
+    // Backward compatibility - previous versions not have the `--version` parameter
+    val oldVersion: String = Try {
+      os.proc(binDirPath / options.binaryName, "about", "--version").call(cwd =
+        os.pwd).out.text.trim
+    }.toOption.getOrElse("0.0.0")
+
+    if (os.exists(binDirPath)) {
+      if (options.force) os.remove.all(binDirPath)
+      else if (newVersion == oldVersion) {
+        System.err.println(
+          s"Scala-cli $newVersion is already installed and up-to-date."
         )
-        val replace = readLine()
-        if (replace == "Y") {
-          os.remove.all(scalaCliBinPath)
+        sys.exit(1)
+      }
+      else if (isOutOfDate(newVersion, oldVersion)) {
+        if (!options.env) println(
+          s"""scala-cli $oldVersion is already installed and out-of-date.
+             |scala-cli will be updated to version $newVersion
+             |""".stripMargin
+        )
+        os.remove.all(binDirPath)
+      }
+      else {
+        if (!options.env && coursier.paths.Util.useAnsiOutput()) {
+          println(s"scala-cli $oldVersion is already installed and up-to-date.")
+          println(s"Do you want to downgrade scala-cli to version $newVersion [Y/n]")
+          val response = readLine()
+          if (response != "Y") {
+            System.err.println("Abort")
+            sys.exit(1)
+          }
+          else
+            os.remove.all(binDirPath)
         }
         else {
-          System.err.println("Abort")
+          System.err.println(
+            s"Error: scala-cli is already installed $oldVersion and up-to-date. Downgrade to $newVersion pass -f or --force."
+          )
           sys.exit(1)
         }
       }
-      else {
-        System.err.println(
-          s"Error: scala-cli already exists. Pass -f or --force to force erasing it."
-        )
-        sys.exit(1)
-      }
     }
 
     os.copy(
-      from = os.Path(options.scalaCliBinaryPath, os.pwd),
-      to = scalaCliBinPath / "scala-cli",
+      from = newScalaCliBinPath,
+      to = binDirPath / options.binaryName,
       createFolders = true
     )
     if (!Properties.isWin)
-      os.perms.set(scalaCliBinPath / "scala-cli", os.PermSet.fromString("rwxrwxr-x"))
-
-    val update = EnvironmentUpdate(Nil, Seq("PATH" -> scalaCliBinPath.toString()))
+      os.perms.set(binDirPath / options.binaryName, os.PermSet.fromString("rwxrwxr-x"))
```

Comment:
This adds permissions for group, user and other. Isn't this too much? How about just +x?